google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITBatchDmlTest.java

```diff
@@ -189,4 +193,47 @@ public void errorBatchDmlAlreadyExist() {
       }
     }
   }
+
+  @Test
+  public void largeBatchDml() {
+    List<Statement> stmts = new LinkedList<>();
+    for (int i = 0; i < 40; i++) {
+      stmts.add(Statement.of("INSERT INTO T (k, v) VALUES ('boo" + i + "', " + i + ");"));
+    }
+
+    for (int i = 0; i < 30; i++) {
+      stmts.add(Statement.of("DELETE FROM T WHERE T.K = 'boo" + i + "';"));
+    }
+
+    final TransactionCallable<long[]> callable = transaction -> transaction.batchUpdate(stmts);
+    TransactionRunner runner = client.readWriteTransaction();
+    long[] rowCounts = runner.run(callable);
+    assertThat(rowCounts.length).isEqualTo(10);
+  }
```

Comment:
Could we also add an assertion for the update count, and not only the length of the update count array?